txmongo/connection.py

```diff
@@ -298,6 +298,10 @@ def __tcp_or_ssl_connect(self, host, port, factory, **kwargs):
             return reactor.connectTCP(host, port, factory, **kwargs)
 
     @property
+    def auth_db(self):
+        return self.__uri['options'].get('authsource') or self.__uri['database']
```

Comment:
This property adds to the public API of Connection without obvious reason. Why do we need this as a public property? Or even if we need it, its name falsely hints that it will return `Database` object, not `str`. My thinking that there is no bold reason to make this public and it is better to make it local variable inside `__init__`.